src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -991,13 +998,32 @@ private async Task<(int, int)> GetInstalledVulnerableAndDeprecatedPackagesCountA
             // Switch off the UI thread before fetching installed packages and deprecation metadata.
             await TaskScheduler.Default;
 
-            PackageCollection installedPackages = await loadContext.GetInstalledPackagesAsync();
-
-            var installedPackageMetadata = await Task.WhenAll(
-                installedPackages.Select(p => GetPackageMetadataAsync(p, packageSources, token)));
-
             int vulnerablePackagesCount = 0;
             int deprecatedPackagesCount = 0;
+            PackageCollection installedPackageCollection = null;
+
+            if (Model.IsSolution)
```

Comment:
A comment clarifying that transitive deps are only displayed in Project-level PM UI would be useful here.

Reply:
Done, thanks!